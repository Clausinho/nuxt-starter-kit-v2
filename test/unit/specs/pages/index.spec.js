// import Vuex from 'vuex';
// import { shallow, createLocalVue } from '@vue/test-utils';
import { shallow } from '@vue/test-utils';
import { createRenderer } from 'vue-server-renderer';
import Page from '~/pages/index.vue';

describe('Page /', () => {
  it('renders to a snapshot', () => {
    const renderer = createRenderer();
    const wrapper = shallow(Page, {});
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});

// const localVue = createLocalVue();

// localVue.use(Vuex);

// describe('Page homepage', () => {
//   let store;
//   const mockLogo = 'mock logo';

//   beforeEach(() => {
//     store = new Vuex.Store({
//       actions: {
//         'logo/logoAction': jest.fn(),
//       },
//       getters: {
//         'logo/logo': () => mockLogo,
//       },
//     });
//   });

//   it('should mock store correctly', () => {
//     const wrapper = shallow(Page, { store, localVue });
//     expect(wrapper.vm.logo).toEqual(mockLogo);
//   });
// });
