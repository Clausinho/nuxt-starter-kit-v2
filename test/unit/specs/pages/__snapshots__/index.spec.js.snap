// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page / renders to a snapshot 1`] = `
<section class="container">
    <img src="~assets/img/logo.png" alt="Nuxt.js Logo" class="logo">
    <h1 class="title">
        USERS
    </h1>
    <ul class="users"></ul>
</section>
`;
